pkg/identity/cache/allocator.go

```diff
@@ -49,8 +49,12 @@ type CachingIdentityAllocator struct {
 
 	identitiesPath string
 
-	events  allocator.AllocatorEventChan
-	watcher identityWatcher
+	// We maintain both a send- and receive-only channel to encode what
+	// operations are allowed in the type system. These must point to the same
+	// underlying channel.
+	eventsRecv allocator.AllocatorEventRecvChan
+	eventsSend allocator.AllocatorEventSendChan
```

Comment:
Honestly, I think I'd just stick to the generic type here and leave a comment that only senders are allowed to close it.

```suggestion
	//  This field exists is to hand out references that are either for sending and receiving.
	// It should not be used directly without converting it first to a allocator.AllocatorEventSendChan
	// or allocator.AllocatorEventRecvChan.
	events allocator.AllocatorEventChan
```

Reply:
yeah, fair. done